packages/semver/src/builders/version/plugin-adapter.ts

```diff
@@ -0,0 +1,22 @@
+import { Plugin, PluginOptions } from './plugin';
+
+export const SUPPORTED_SEMANTIC_RELEASE_PLUGINS = ['@semantic-release/npm'];
+
+export interface SemanticReleasePlugin extends Plugin {
+  addChannel?(options: PluginOptions): Promise<unknown>;
+}
+
+export class SemanticReleasePluginAdapter implements Plugin {
+  constructor(private _plugin: SemanticReleasePlugin) {}
+
+  publish(options: PluginOptions) {
+    this._plugin.addChannel(options);
+    return this._plugin.publish(options);
+  }
+}
+
+export function _adapt(pluginName: string, plugin: Plugin): Plugin {
+  return SUPPORTED_SEMANTIC_RELEASE_PLUGINS.includes(pluginName)
```

Comment:
🤔 oh! I didn't think about a whitelist. How do we make sure that we are not calling a `semantic-release` plugin? Well, in the worst case, it will just crash 😊
I was imagining something that would detect if it's a `semantic-release` plugin or not.
By the way, as both `semver` and `semantic-release` plugins have a `publish` function but with a different signature, we can't detect the plugin using duck typing.

Maybe we could provide a `@jscutlery/semver-plugin-utils` util with a `createSemverPlugin` function that returns an object with some magic string in it: `{publish() {}, name: 'my-plugin', type: '@jscutlery/semver-plugin'}`.
We can check the `@jscutlery/semver-plugin` magic string.

This is strict but cumbersome as plugin developers will have to depend on `@jscutlery/semver-plugin-utils`... which might be cool as it could contain other helpers in the future.

If we don't do this from the beginning, we will have some plain object plugins in the wild and we wouldn't be able to verify the magic string as it would be a breaking change.